examples/flutter_gallery/lib/demo/rally/finance.dart

```diff
@@ -0,0 +1,329 @@
+    subtitle: amountUsed + ' / ' + primaryAmount,
+    indicatorColor: RallyColors.budgetColor(budgetDataIndex),
+    indicatorFraction: item.amountUsed / item.primaryAmount,
+    amount: item.primaryAmount - item.amountUsed,
+  );
+}
+
+List<FinancialEntityCategoryView> buildAccountDataListViews(
+    List<AccountData> items) {
+  return List<FinancialEntityCategoryView>.generate(
+    items.length,
+    (int i) => buildFinancialEntityFromAccountData(items[i], i),
+  );
+}
+
+List<FinancialEntityCategoryView> buildBillDataListViews(List<BillData> items) {
+  return List<FinancialEntityCategoryView>.generate(
+    items.length,
+    (int i) => buildFinancialEntityFromBillData(items[i], i),
+  );
+}
+
+List<FinancialEntityCategoryView> buildBudgetDataListViews(
+    List<BudgetData> items, BuildContext context) {
+  return <FinancialEntityCategoryView>[
+    for (int i = 0; i < items.length; i++)
+      buildFinancialEntityFromBudgetData(items[i], i, context)
+  ];
+}
+
+class FinancialEntityCategoryDetailsPage extends StatelessWidget {
+  final List<DetailedEventData> items = DummyDataService.getDetailedEventItems();
+
+  @override
+  Widget build(BuildContext context) {
+    final List<_DetailedEventCard> cards = items.map((DetailedEventData detailedEventData) {
+      return _DetailedEventCard(
+        title: detailedEventData.title,
+        subtitle: dateFormat.format(detailedEventData.date),
+        amount: detailedEventData.amount,
+      );
+    }).toList();
+
+    return Scaffold(
+      appBar: AppBar(
+        elevation: 0,
+        centerTitle: true,
+        title: Text(
+          'Checking',
+          style: Theme.of(context).textTheme.body1.copyWith(fontSize: 18),
+        ),
+      ),
+      body: Column(
+        children: <Widget>[
+          SizedBox(
+            height: 200,
+            width: double.infinity,
+            child: RallyLineChart(events: items),
+          ),
+          Flexible(
+            child: ListView(shrinkWrap: true, children: cards),
+          ),
+        ],
+      ),
+    );
+  }
+}
+
+class _DetailedEventCard extends StatelessWidget {
+  const _DetailedEventCard({
+    @required this.title,
+    @required this.subtitle,
+    @required this.amount,
+  });
+
+  final String title;
+  final String subtitle;
+  final double amount;
+
+  @override
+  Widget build(BuildContext context) {
+    return FlatButton(
+      onPressed: () {},
+      child: SizedBox(
+        height: 68,
+        child: Column(
+          children: <Widget>[
+            SizedBox(
+              height: 67,
+              child: Row(
+                children: <Widget>[
+                  Column(
+                    mainAxisAlignment: MainAxisAlignment.center,
+                    crossAxisAlignment: CrossAxisAlignment.start,
+                    children: <Widget>[
+                      Text(
+                        title,
+                        style: Theme.of(context).textTheme.body1.copyWith(fontSize: 16),
```

Comment:
Maybe bind textTheme to a variable at the top of the build method

Reply:
I agree, done!